projects/ngx-qgrid-plugins/src/lib/cell-tooltip/cell-tooltip.component.ts

```diff
@@ -0,0 +1,72 @@
+import {
+	ChangeDetectorRef,
+	ChangeDetectionStrategy,
+	Component,
+	OnChanges,
+	Input,
+	ApplicationRef,
+} from '@angular/core';
+import { DomTd, GridPlugin, TemplateHostService } from '@qgrid/ngx';
+
+@Component({
+	selector: 'q-grid-cell-tooltip',
+	templateUrl: './cell-tooltip.component.html',
+	providers: [GridPlugin, TemplateHostService],
+	changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class CellTooltipComponent implements OnChanges {
+	@Input() showDelay: number;
+	private cell: HTMLElement;
+	context: { $implicit: DomTd } = {
+		$implicit: null,
+	};
+
+	constructor(
+		private plugin: GridPlugin,
+		private cd: ChangeDetectorRef,
+		private appRef: ApplicationRef
+	) {}
+
+	ngOnChanges() {
+		const { model, observe, table } = this.plugin;
+		observe(model.highlightChanged).subscribe((e) => {
+			if (e.hasChanges('cell') && e.state.cell) {
+				const domCell = table.body.cell(
+					e.state.cell.rowIndex,
+					e.state.cell.columnIndex
+				);
+				if (domCell) {
```

Comment:
it's guarantee that domCell is not null, but it's not guaranteed that domCell.model() is not null